chesster.js

```diff
@@ -1169,3 +1177,71 @@ controller.hears([
         }
     });
 });
+
+
+/* Scheduling */
+
+// reply to any incoming message
+controller.on('ambient', function(bot, message) {
+    bot_exception_handler(bot, message, function(){
+        var channel = channels.byId[message.channel];
+        var scheduling_options = config.scheduling[channel.name];
+        if (!scheduling_options) {
+            return;
+        } 
+        try {
+            var results = spreadsheets.parse_scheduling(message.text, {
+                offset_hours: scheduling_options.offset_hours
+            });
+            var white = users.getByNameOrID(results.white);
+            var black = users.getByNameOrID(results.black);
+            results.white = white.name;
+            results.black = black.name;
+            spreadsheets.update_schedule(
+                config.service_account_auth,
+                scheduling_options.key,
+                scheduling_options.colname,
+                scheduling_options.format,
+                results,
+                function(err, reversed) {
+                    if (err) {
+                        if (err.indexOf && err.indexOf("Unable to find pairing.") == 0) {
+                            user = "<@"+message.user+">";
+                            bot.reply(message, ":x: " + user + " I couldn't find your pairing. Please use a format like: @lakinwecker v @lakinwecker 04/16 @ 16:00 GMT");
+                        } else {
+                            bot.reply(message, "Something went wrong. Notify @lakinwecker");
+                            throw new Error("Error updating scheduling sheet: " + err);
+                        }
+                    } else {
+                        if (reversed) {
+                            var tmp = white;
+                            white = black;
+                            black = tmp;
+                        }
+                        var whiteDate = results.date.clone().utcOffset(white.tz_offset/60);
```

Comment:
I really like this. I think it will make things more clear for BOTH parties.

I see why you need white/black above


Ignore my previous comment